### ISSUE TITLE
Cloud Events (Binary Content Mode) cause json unmarshal error in event controller.  

### ISSUE THREAD

**Issue by rbaethe** [kyma-project/kyma#12093]
**Description**

Cloud Events (Binary Content Mode) cause json unmarshal error in event controller.  

Kyma Open Source: 1.24

**Expected result**

Sent event is successfully processed and forwarded to subscriber. 

**Actual result**

Event controller throws error. Event does not reach event subscriber.
```json
{
  "level": "ERROR",
  "timestamp": "2021-09-08T11:29:35Z",
  "logger": "reconciler.Subscription",
  "caller": "handlers/nats.go:167",
  "message": "failed to convert Nats message to CE",
  "context": {
    "error": "json: cannot unmarshal object into Go value of type string"
  }
}
```
**Steps to reproduce**

Send event in binary content mode 

`curl -k -i \
    --data-raw '{"more": "banane"}' \
    -H "Content-Type: application/json; charset=utf-8" \
    -H "ce-specversion: 1.0" \
    -H "ce-type: sap.kyma.custom.slm.slmproduct.published.v1" \
    -H "ce-eventtypeversion: v1" \
    -H "ce-id: e15caa09-98ff-4973-be42-fb685f3e3ef2" \
    -H "ce-source: kyma" \
    "http://eventing-event-publisher-proxy.kyma-system/publish"`

Please note, the same event in structured content mode is processed succesfully:

`curl -k -i \
    --data @<(cat <<EOF
    {
        "source": "slm",
        "specversion": "1.0",
        "eventtypeversion": "v1",
        "data": {"more":"banane"},
        "datacontenttype": "application/json",
        "id": "759815c3-b142-48f2-bf18-c6502dc0998f",
        "type": "sap.kyma.custom.slm.slmproduct.published.v1"
    }
EOF
    ) \
    -H "Content-Type: application/cloudevents+json" \
    "http://eventing-event-publisher-proxy.kyma-system/publish"`

**Troubleshooting**

Build custom converter to convert events from binary to structured mode.

**Comment by radufa** [comment]
Tested with the Kyma main branch nd it works.

Steps to test it:
1. Create the namspace test:
```
kuvbectl create namespace test
```
2. Create a function:
```
cat << EOF | k apply -f -
apiVersion: serverless.kyma-project.io/v1alpha1
kind: Function
metadata:
  generation: 2
  name: test
  namespace: test
spec:
  buildResources:
    limits:
      cpu: 1700m
      memory: 1700Mi
    requests:
      cpu: 1100m
      memory: 1100Mi
  deps: "{ \n  \"name\": \"test\",\n  \"version\": \"1.0.0\",\n  \"dependencies\":
    {}\n}"
  maxReplicas: 5
  minReplicas: 1
  resources:
    limits:
      cpu: 200m
      memory: 256Mi
    requests:
      cpu: 100m
      memory: 128Mi
  runtime: nodejs12
  source: "module.exports = { \n  main: function (event, context) {\n    console.log(event.data);\n
    \   return \"Hello World!\";\n  }\n}"
EOF
```
- Tail the log of this function

3. Create a subscription:
```
cat << EOF | k apply -f -
apiVersion: eventing.kyma-project.io/v1alpha1
kind: Subscription
metadata:
  name: test
  namespace: test
spec:
  filter:
    filters:
    - eventSource:
        property: source
        type: exact
        value: ""
      eventType:
        property: type
        type: exact
        value: sap.kyma.custom.noapp.order.created.v1
  protocol: ""
  protocolsettings: {}
  sink: http://test.test.svc.cluster.local
EOF
```
4.  Port-forward publisher-proxy:
```
kubectl -n kyma-system port-forward svc/eventing-event-publisher-proxy  8081:80
```
2. the Send the cloud event in binary mode:
```
curl -k -i \
    --data-raw '{"more": "banane"}' \
    -H "Content-Type: application/json; charset=utf-8" \
    -H "ce-specversion: 1.0" \
    -H "ce-type: sap.kyma.custom.noapp.order.created.v1" \
    -H "ce-eventtypeversion: v1" \
    -H "ce-id: e15caa09-98ff-4973-be42-fb685f3e3ef2" \
    -H "ce-source: kyma" \
    http://localhost:8081/publish
```

**Comment by radufa** [comment]
The issue was reproduced in the last Kyma release 1.24.6

**Comment by radufa** [comment]
Working to fix it for the next Kyma release 1.24.7
